### DISCUSSION TITLE
How to delete function metadata?

### DISCUSSION THREAD

**Discussion by imnuke** [naim94a/lumen#144] [Q&A] [answered]
I've enabled `allow_deletes` in the config file, but I can't find a button in my IDA Pro 8.4 that allows me to delete my pushed metadata. Anybody can help me?

![图片](https://github.com/naim94a/lumen/assets/17084130/02531f4c-a32b-4187-bcdb-994a1c0adfa5)

![图片](https://github.com/naim94a/lumen/assets/17084130/a0458914-94d9-4750-83a9-c847daee74cd)

**Comment by tomrus88** [comment, marked as answer]
You must have special (admin) features enabled on account in order for those options to show up. You can access them in each function context menu (Lumina -> Metadata History, new window will open, in that window right click on a function in the list and it should have Delete menu item, which only appears if you have metadata for this function pulled from server).

Or you can use **lc** (lumina client) command like tool that comes with official limina server installer, if you have access to that ofc...

![image_2024-03-15_081006048](https://github.com/naim94a/lumen/assets/12560/c51e16a3-b706-4417-9c6d-a4badc7aa17e)

**Reply by imnuke** [reply]
Thank you. But how can I enable the admin features on my account? I use lumen to setup the lumina server, I don't have the "Metadata History" entry in my "Lumina" menu.

**Reply by tomrus88** [reply]
Metadata history menu appears when you open context menu on specific function, it's not available in main Lumina menu.

![image_2024-03-15_195319047](https://github.com/naim94a/lumen/assets/12560/a9a911ce-baaf-4afb-922d-b13c8fe2ee8b)

**Reply by tomrus88** [reply]
This was recently fixed and should work in IDA Pro 8.3+

**Reply by imnuke** [reply]
Thanks for your reply and fix, but it seems still not working in my IDA Pro 8.4. I've pulled the latest version of this repository with `git pull` and then rebuilt the containers with `sudo docker compose down && sudo docker compose up --build`. And I've tried restarting my IDA Pro. There are still no "Delete function XXXX" in my Lumina metadata window.

![图片](https://github.com/naim94a/lumen/assets/17084130/c754af14-2958-4704-ad70-23709baf6bb0)

I want to help but I'm not familiar with lumina protocol or rust language. Can I provide some debugging information that might help?

**Reply by naim94a** [reply]
Make sure it's enabled in the configuration file (`lumina.allow_deletes`). Also, the option should be in the context menu of the function (right click on `output_json` on the left)

**Reply by imnuke** [reply]
Thank you! I find it in the context menu of the function! But it only appears for function that pulled from the server (for example, I push the output_json function, and then pull it, so the function becomes green-backgrounded), if I just push the function, the entry still not appears.

![图片](https://github.com/naim94a/lumen/assets/17084130/9ef9bba4-d9d7-4ea2-b8ae-32e00152e66b)

![图片](https://github.com/naim94a/lumen/assets/17084130/253063c8-2880-4586-8030-75c21bbddeb6)

![图片](https://github.com/naim94a/lumen/assets/17084130/76e529dc-f9e7-491a-8dc0-7f33382993cd)

Is it a bug or it's just a feature?

**Reply by tomrus88** [reply]
That's just how it works, only pulled functions have Delete menu item.

**Reply by imnuke** [reply]
Ok, thank you!
